src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandleExtensions.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Provides extension methods to operate with GC handles.
+    /// </summary>
+    public static class GCHandleExtensions
```

Comment:
Why are these extension methods? They are defined in the same namespace in the same assembly. I don't understand the utility for these being extensions methods in this case as opposed to member/static functions on the actual type.

Reply:
I imagine it's so that they can look like instance members, but only apply to specific `T`s (currently a `T[]` or `string`).

Reply:
Yes, it's used effectively as generic specification. Of course with proper generic specification we can avoid dealing with nullability of generic parameter of `this`.